components/brave_new_tab_ui/components/default/cryptoDotCom/index.tsx

```diff
@@ -0,0 +1,459 @@
+
+import {
+  ActionAnchor,
+  ActionButton,
+  AssetIcon,
+  AssetIconWrapper,
+  BackArrow,
+  BasicBox,
+  Box,
+  CryptoDotComIcon,
+  FlexItem,
+  Header,
+  List,
+  ListItem,
+  PlainButton,
+  StyledTitle,
+  StyledTitleText,
+  SVG,
+  Text,
+  WidgetWrapper,
+  UpperCaseText
+} from './style'
+import CryptoDotComLogo from './assets/cryptoDotCom-logo'
+import { CaratLeftIcon } from 'brave-ui/components/icons'
+
+// Utils
+import cryptoColors from '../exchangeWidget/colors'
+import { getLocale } from '../../../../common/locale'
+
+interface State {
+  selectedAsset: string
+}
+
+interface TickerPrice {
+  price: number
+  volume: number
+}
+
+interface AssetRanking {
+  lastPrice: number
+  pair: string
+  percentChange: number
+}
+
+interface ChartDataPoint {
+  c: number
+  h: number
+  l: number
+  o: number
+  t: number
+  v: number
+}
+
+interface Props {
+  showContent: boolean
+  optInTotal: boolean
+  optInBTCPrice: boolean
+  optInMarkets: boolean
+  tickerPrices: Record<string, TickerPrice>
+  losersGainers: Record<string, AssetRanking[]>
+  supportedPairs: Record<string, string[]>
+  charts: Record<string, ChartDataPoint[]>
+  stackPosition: number
+  onShowContent: () => void
+  onDisableWidget: () => void
+  onTotalPriceOptIn: () => void
+  onBtcPriceOptIn: () => void
+  onSetLosersGainers: () => Promise<void>
+  onSetSupportedPairs: () => Promise<void>
+  onSetTickerPrices: (assets: string[]) => Promise<void>
+  onSetCharts: (asset: string[]) => Promise<void>
+  onUpdateActions: () => Promise<void[]>
+  onBuyCrypto: () => void
+  onInteraction: () => void
+  onOptInMarkets: () => void
+}
+interface ChartConfig {
+  data: Array<any>
+  chartHeight: number
+  chartWidth: number
+}
+
+class CryptoDotCom extends React.PureComponent<Props, State> {
+  private refreshInterval: any
+  private topMovers: string[] = [ 'BTC', 'ETH', 'CRO' ]
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      selectedAsset: ''
+    }
+  }
+
+  // This is a temporary function only necessary for MVP
+  // Merges losers/gainers into one table
+  transformLosersGainers = ({ losers = [], gainers = [] }: Record<string, AssetRanking[]>): Record<string, AssetRanking> => {
+    const losersGainersMerged = [ ...losers, ...gainers ]
+    return losersGainersMerged.reduce((mergedTable: object, asset: AssetRanking) => {
+      let { pair: assetName, ...assetRanking } = asset
+      assetName = assetName.split('_')[0]
+
+      return {
+        ...mergedTable,
+        [assetName]: assetRanking
+      }
+    }, {})
+  }
+
+  componentDidMount () {
+    this.checkSetRefreshInterval()
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.refreshInterval)
+  }
+
+  checkSetRefreshInterval = () => {
+    if (!this.refreshInterval) {
+      this.refreshInterval = setInterval(async () => {
+        await this.props.onUpdateActions()
+          .catch((_e) => console.debug('Could not update crypto.com data'))
+      }, 30000)
```

Comment:
nit: this timer could be firing in some other class handling redux actions. Fine here, but IMO we should be thinking about keeping these react components simple and putting logic in to sync and async redux reducers.